roles/hostedbm/tasks/create-cluster.yml

```diff
@@ -0,0 +1,130 @@
+---
+- name: Run hcp create cluster agent --render
+  environment:
+    KUBECONFIG: "{{ hostedbm_kubeconfig_file | default(omit) }}"
+  ansible.builtin.command: >
+    {{ hostedbm_working_dir }}/hcp create cluster agent
+    --pull-secret={{ hostedbm_working_dir }}/openshift_pull.json
+    --name={{ hostedbm_cluster_name }}
+    --agent-namespace={{ hostedbm_agent_ns }}
+    --node-pool-replicas={{ hostedbm_node_pool_replicas }}
+    --base-domain={{ hostedbm_cluster_base_domain }}
+    --etcd-storage-class={{ hostedbm_storage_class }}
+    --ssh-key={{ hostedbm_working_dir }}/id_rsa.pub
+    --control-plane-availability-policy={{ hostedbm_availability_policy }}
+    --release-image={{ hostedbm_release_image }}
+    --annotations hypershift.openshift.io/skip-release-image-validation=true
+    {% if hostedbm_bm_cpo_override_image is defined and hostedbm_bm_cpo_override_image | length > 0 %}
+    --annotations hypershift.openshift.io/control-plane-operator-image={{ hostedbm_bm_cpo_override_image }}{% endif %}
+    --namespace={{ hostedbm_cluster_ns }}
+    --render
+  register: _hostedbm_hcp_output
+
+- name: Save output to YAML file
+  ansible.builtin.copy:
+    content: "{{ _hostedbm_hcp_output.stdout }}"
+    dest: "{{ hostedbm_working_dir }}/hosted_cluster_rendered.yaml"
+    mode: "0644"
+
+# replace nodeport with loadbalancer
+- name: Run Bash script to update Kubeapiservice
+  ansible.builtin.command: >
```

Comment:
Why not using: `ansible.builtin.replace` ?

Reply:
@reihl 

I tried various combination with replace and lineinfile, But I couldn't manage to do the below with ansible replace:-

~~~
sed -i '/^\s*- service: APIServer$/,/^\s*type: NodePort$/{s/^\(\s*type:\s*\)NodePort$/\1LoadBalancer/; /^\s*nodePort:/, /^\s*address:/d}' myorig.yaml
~~~

cat myorig.yaml
~~~
---
apiVersion: hypershift.openshift.io/v1beta1
kind: HostedCluster
metadata:
  name: testing
  namespace: clusters
spec:
  services:
  - service: APIServer
    servicePublishingStrategy:
      nodePort:
        address: 192.168.98.73
      type: NodePort
  - service: OAuthServer
    servicePublishingStrategy:
      nodePort:
        address: 192.168.98.73
      type: NodePort
  - service: OIDC
    servicePublishingStrategy:
      nodePort:
        address: 192.168.98.73
      type: None
  - service: Konnectivity
    servicePublishingStrategy:
      nodePort:
        address: 192.168.98.73
      type: NodePort
  - service: Ignition
    servicePublishingStrategy:
      nodePort:
        address: 192.168.98.73
      type: NodePort
  - service: OVNSbDb
    servicePublishingStrategy:
      nodePort:
        address: 192.168.98.73
      type: NodePort
  sshKey:
    name: cnfdr36-ssh-key
~~~


After modification:-
~~~
---
apiVersion: hypershift.openshift.io/v1beta1
kind: HostedCluster
metadata:
  name: testing
  namespace: clusters
spec:
  services:
  - service: APIServer
    servicePublishingStrategy:
      type: LoadBalancer
  - service: OAuthServer
    servicePublishingStrategy:
      nodePort:
        address: 192.168.98.73
      type: NodePort
  - service: OIDC
    servicePublishingStrategy:
      nodePort:
        address: 192.168.98.73
      type: None
  - service: Konnectivity
    servicePublishingStrategy:
      nodePort:
        address: 192.168.98.73
      type: NodePort
  - service: Ignition
    servicePublishingStrategy:
      nodePort:
        address: 192.168.98.73
      type: NodePort
  - service: OVNSbDb
    servicePublishingStrategy:
      nodePort:
        address: 192.168.98.73
      type: NodePort
  sshKey:
    name: cnfdr36-ssh-key
~~~


If you can help with code how to achieve above via ansible, I am happy to replace the code with native ansible module.

Reply:
Thank you so much!